### ISSUE TITLE
[BUG] Dask groupby + reset_index fails when group key is a string column

### ISSUE THREAD

**Issue by ayushdg** [rapidsai/cudf#4252]
**Describe the bug**
Groupby + reset_index on dask_cudf dataframes fails if the group_key is a string column

**Steps/Code to reproduce bug**
```python
df = cudf.datasets.randomdata(nrows=10,dtypes={'a':str,'b':int,'c':int})
ddf = dask_cudf.from_cudf(df,2)
res = ddf.groupby('a', sort=True).sum()
res.reset_index().head()

ValueError: The columns in the computed data do not match the columns in the provided metadata
  Extra:   ['a']
  Missing: ['index']
```
<details>
<summary> Error Trace </summary>
<pre>
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-34-7ce80e804ebc> in <module>
----> 1 res.reset_index().head()

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/core.py in head(self, n, npartitions, compute)
    969             Whether to compute the result, default is True.
    970         """
--> 971         return self._head(n=n, npartitions=npartitions, compute=compute, safe=True)
    972 
    973     def _head(self, n, npartitions, compute, safe):

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/core.py in _head(self, n, npartitions, compute, safe)
   1002 
   1003         if compute:
-> 1004             result = result.compute()
   1005         return result
   1006 

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/base.py in compute(self, **kwargs)
    164         dask.base.compute
    165         """
--> 166         (result,) = compute(self, traverse=False, **kwargs)
    167         return result
    168 

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/base.py in compute(*args, **kwargs)
    435     keys = [x.__dask_keys__() for x in collections]
    436     postcomputes = [x.__dask_postcompute__() for x in collections]
--> 437     results = schedule(dsk, keys, **kwargs)
    438     return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])
    439 

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/local.py in get_sync(dsk, keys, **kwargs)
    525     """
    526     kwargs.pop("num_workers", None)  # if num_workers present, remove it
--> 527     return get_async(apply_sync, 1, dsk, keys, **kwargs)
    528 
    529 

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/local.py in get_async(apply_async, num_workers, dsk, result, cache, get_id, rerun_exceptions_locally, pack_exception, raise_exception, callbacks, dumps, loads, **kwargs)
    492 
    493                 while state["ready"] and len(state["running"]) < num_workers:
--> 494                     fire_task()
    495 
    496             succeeded = True

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/local.py in fire_task()
    464                         pack_exception,
    465                     ),
--> 466                     callback=queue.put,
    467                 )
    468 

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/local.py in apply_sync(func, args, kwds, callback)
    514 def apply_sync(func, args=(), kwds={}, callback=None):
    515     """ A naive synchronous version of apply_async """
--> 516     res = func(*args, **kwds)
    517     if callback is not None:
    518         callback(res)

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/local.py in execute_task(key, task_info, dumps, loads, get_id, pack_exception)
    225         failed = False
    226     except BaseException as e:
--> 227         result = pack_exception(e, dumps)
    228         failed = True
    229     return key, result, failed

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/local.py in execute_task(key, task_info, dumps, loads, get_id, pack_exception)
    220     try:
    221         task, data = loads(task_info)
--> 222         result = _execute_task(task, data)
    223         id = get_id()
    224         result = dumps((result, id))

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/core.py in _execute_task(arg, cache, dsk)
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]
--> 118         args2 = [_execute_task(a, cache) for a in args]
    119         return func(*args2)
    120     elif not ishashable(arg):

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/core.py in <listcomp>(.0)
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]
--> 118         args2 = [_execute_task(a, cache) for a in args]
    119         return func(*args2)
    120     elif not ishashable(arg):

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/core.py in _execute_task(arg, cache, dsk)
    117         func, args = arg[0], arg[1:]
    118         args2 = [_execute_task(a, cache) for a in args]
--> 119         return func(*args2)
    120     elif not ishashable(arg):
    121         return arg

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/optimization.py in __call__(self, *args)
    980         if not len(args) == len(self.inkeys):
    981             raise ValueError("Expected %d args, got %d" % (len(self.inkeys), len(args)))
--> 982         return core.get(self.dsk, self.outkey, dict(zip(self.inkeys, args)))
    983 
    984     def __reduce__(self):

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/core.py in get(dsk, out, cache)
    147     for key in toposort(dsk):
    148         task = dsk[key]
--> 149         result = _execute_task(task, cache)
    150         cache[key] = result
    151     result = _execute_task(out, cache)

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/core.py in _execute_task(arg, cache, dsk)
    117         func, args = arg[0], arg[1:]
    118         args2 = [_execute_task(a, cache) for a in args]
--> 119         return func(*args2)
    120     elif not ishashable(arg):
    121         return arg

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/utils.py in apply(func, args, kwargs)
     27 def apply(func, args, kwargs=None):
     28     if kwargs:
---> 29         return func(*args, **kwargs)
     30     else:
     31         return func(*args)

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/core.py in apply_and_enforce(*args, **kwargs)
   5050             return meta
   5051         if is_dataframe_like(df):
-> 5052             check_matching_columns(meta, df)
   5053             c = meta.columns
   5054         else:

/opt/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/utils.py in check_matching_columns(meta, actual)
    676             " the columns in the provided metadata\n"
    677             "  Extra:   %s\n"
--> 678             "  Missing: %s" % (extra, missing)
    679         )
    680 

ValueError: The columns in the computed data do not match the columns in the provided metadata
  Extra:   ['a']
  Missing: ['index']
</pre>
</details>

**Expected behavior**
No error

**Environment overview (please complete the following information)**
 - Environment location: Docker:  Latest docker dev nightly (Feb25)
 - Method of cuDF install: Cudf at 4460093650932fbf476dfd16194c51b07755325e

**Additional context**

The index name seems to be lost in `_meta_nonempty`

```
res.index.name
'a'
```
```
res._meta
```

  | b | c
--| -- | --
a | |

```
res._meta_nonempty

        b	c
cat	0	0
dog	1	1
```

```
res.reset_index().columns
Index(['index', 'b', 'c'], dtype='object')
# Should probably be 'a','b','c'
```

**Comment by kkraus14** [comment]
@ayushdg are you on the latest cuDF build? This sounds very similar to what was fixed in #4221

**Comment by VibhuJawa** [comment]
Can repro this behavior on below (which is the latest i think) : 

```
# packages in environment at /raid/vjawa/conda_install/conda_env/envs/cudf_25_feb:
cudf                      0.13.0a200225         py37_2550    rapidsai-nightly
dask-cudf                 0.13.0a200225         py37_2550    rapidsai-nightly
libcudf                   0.13.0a200225     cuda10.1_2550    rapidsai-nightly
```

**Comment by ayushdg** [comment]
> @ayushdg are you on the latest cuDF build?
I'm at 4460093 (which is from the latest nightly). 

> This sounds very similar to what was fixed in #4221

They are very similar. This one is specific to string columns as group keys and is independent of the `sort` behavior within groupby

**Comment by kkraus14** [comment]
Building with the latest of 0.13 to confirm I can reproduce, will then dig in.
